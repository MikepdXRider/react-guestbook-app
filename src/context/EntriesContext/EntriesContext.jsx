import React, { createContext, useContext, useState } from 'react'


const EntriesContext = createContext();

const EntriesProvider = ({children}) => {
    const [entries, setEntries] = useState([]);

    return(
        <EntriesContext.Provider value={{entries, setEntries}}>
            {children}
        </EntriesContext.Provider>
    )
}

const useEntries = () => {
    const context = useContext(EntriesContext);
    
    if (context === undefined) throw new Error('useEntries must be declared within an EntriesProvider');
    
    return context;
}

export { EntriesProvider, useEntries }