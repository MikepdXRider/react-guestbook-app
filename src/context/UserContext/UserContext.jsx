import React, { createContext, useContext, useState } from 'react'


const UserContext = createContext();

const UserProvider = ({children}) => {
    const [username, setUsername] = useState('');

    return(
        <UserContext.Provider value={{username, setUsername}}>
            {children}
        </UserContext.Provider>
    )
}


// 
// Could move to useAuth.jsx in the hooks folder. 
// 
const useUser = () => {
    const context = useContext(UserContext);

    if (context === undefined) throw new Error('useUser must be declared within an UserProvider');

    return context;
}

export { UserProvider, useUser }