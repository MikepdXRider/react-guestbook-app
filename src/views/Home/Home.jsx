import React from 'react'
import EntryList from '../../components/Entries/EntryList'
import EntryForm from '../../components/Forms/EntryForm'
import NameForm from '../../components/Forms/NameForm.jsx'
import { useUser } from '../../context/UserContext/UserContext.jsx'

// This will be the primary landing page of the app. It will contain all of the components/elements that need to be rendered to the page(forms, entries, etc..)
export default function Home() {
    const {username, setUsername} = useUser();

    function handleLogOut() {
        setUsername('');
    }

    return (
        <main>
            <section>
            {
                username 
                ? <div>
                    <p>Signed in as {username}</p>
                    <button onClick={() => handleLogOut()}>log out</button>
                </div>
                : <NameForm />
            }
            </section>

                <EntryForm />
                <EntryList />
        </main>
    )
}
