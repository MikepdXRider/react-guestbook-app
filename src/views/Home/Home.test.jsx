import react from "react";
import { render } from '@testing-library/react';
import { UserProvider } from '../../context/UserContext/UserContext.jsx';
import { EntriesProvider } from '../../context/EntriesContext/EntriesContext.jsx';
import Home from "./Home.jsx";

it('renders the home view', () => {
    const {container} = render(
            <UserProvider>
                <EntriesProvider>
                    <Home />
                </EntriesProvider>
            </UserProvider>
    );

    expect(container).toMatchSnapshot();
});