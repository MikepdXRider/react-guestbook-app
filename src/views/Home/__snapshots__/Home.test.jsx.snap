// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the home view 1`] = `
<div>
  <main>
    <section>
      <form>
        <fieldset>
          <legend>
            Name
          </legend>
          <input
            type="text"
            value=""
          />
          <button>
            Add Name
          </button>
        </fieldset>
      </form>
    </section>
    <form>
      <fieldset>
        <legend>
          Entry
        </legend>
        <input
          type="text"
          value=""
        />
        <button>
          Add Entry
        </button>
      </fieldset>
    </form>
    <section>
      <p>
        Entries
      </p>
    </section>
  </main>
</div>
`;
